relayer/chains/cosmos/tx.go

```diff
@@ -1230,6 +1231,28 @@ func (cc *CosmosProvider) queryTMClientState(ctx context.Context, srch int64, sr
 	return clientState, nil
 }
 
+// queryLocalhostClientState retrieves the latest consensus state for a client in state at a given height
+// and unpacks/cast it to localhost client state.
+func (cc *CosmosProvider) queryLocalhostClientState(ctx context.Context, srch int64) (*localhost.ClientState, error) {
+	clientStateRes, err := cc.QueryClientStateResponse(ctx, srch, ibcexported.LocalhostClientID)
+	if err != nil {
+		return &localhost.ClientState{}, err
+	}
+
+	clientStateExported, err := clienttypes.UnpackClientState(clientStateRes.ClientState)
+	if err != nil {
+		return &localhost.ClientState{}, err
+	}
+
+	clientState, ok := clientStateExported.(*localhost.ClientState)
+	if !ok {
+		return &localhost.ClientState{},
+			fmt.Errorf("error when casting exported clientstate to localhost client type")
```

Comment:
Use `%T` to include the actual type that failed, to simplify debugging if this condition is ever reached.